Enable Minitest/LiteralAsActualArgument

shouichi opened the pull request:
There are assertions that expected/actual arguments are passed in the
reversed order by mistake. Enabling the LiteralAsActualArgument rule
prevents this mistake from happening.

The existing tests were auto-corrected by rubocop with a bit of
indentation adjustment.

shouichi commented on actionpack/test/controller/parameters/mutators_test.rb:
```diff
@@ -181,7 +181,7 @@ class ParametersMutatorsTest < ActiveSupport::TestCase
     params.permit!
     params_hash = params.to_h { |key, value| [:"#{key}_modified", value] }
-    assert_equal params_hash.keys, %w(name_modified age_modified location_modified)
+    assert_equal(%w(name_modified age_modified location_modified), params_hash.keys)
```
Should I remove parentheses to be consistent with other tests in the same file?

jonathanhefner replied:
I think if there were no parentheses before, we should not add them.  And if there were parentheses before, then we should keep them.

Are these being parentheses from Rubocop's autocorrect?  If so, that is unfortunate. :disappointed:

shouichi replied:
@koic Thank you for the fix in https://github.com/rubocop/rubocop-minitest/pull/259! I tried that branch and it works almost perfectly except for the below corner case.

```
# original
assert_equal @credentials.config, {}

# auto-corrected: syntax error because the {} is interpreted as a block.
assert_equal {}, @credentials.config
```

jonathanhefner commented on activerecord/test/cases/adapters/abstract_mysql_adapter/connection_test.rb:
```diff
@@ -200,8 +200,7 @@ def test_get_and_release_advisory_lock
 
-    assert_equal test_lock_free(lock_name), false,
-      "expected the test advisory lock to be held but it wasn't"
+    assert_equal(false, test_lock_free(lock_name), "expected the test advisory lock to be held but it wasn't")
```
Let's keep the error message on a separate line like before:

```suggestion
    assert_equal false, test_lock_free(lock_name),
      "expected the test advisory lock to be held but it wasn't"
```

koic replied:
> Are these being parentheses from Rubocop's autocorrect? If so, that is unfortunate.

Ah, this should ideally be handled by RuboCop Minitest, but it seems to produce unexpected auto-correction code. I will fix this autocorrection logic of RuboCop Minitest.

jonathanhefner commented on activerecord/test/cases/adapters/abstract_mysql_adapter/connection_test.rb:
```diff
@@ -212,8 +211,7 @@ def test_get_and_release_advisory_lock
     released_non_existent_lock = @connection.release_advisory_lock(lock_name)
-    assert_equal released_non_existent_lock, false,
-      "expected release_advisory_lock to return false when there was no lock to release"
+    assert_equal(false, released_non_existent_lock, "expected release_advisory_lock to return false when there was no lock to release")
```
Here too, let's keep the error message on a separate line like before:

```suggestion
    assert_equal false, released_non_existent_lock,
      "expected release_advisory_lock to return false when there was no lock to release"
```

jonathanhefner commented on activemodel/test/cases/callbacks_test.rb:
```diff
@@ -55,33 +55,33 @@ def create
     model.create
-    assert_equal model.callbacks, [ :before_create, :before_around_create, :create,
-                                    :after_around_create, :after_create, :final_callback]
+    assert_equal([ :before_create, :before_around_create, :create,
+                                    :after_around_create, :after_create, :final_callback], model.callbacks)
```
The indentation here is a bit odd.  How about:

```suggestion
    assert_equal \
      [:before_create, :before_around_create, :create, :after_around_create, :after_create, :final_callback],
      model.callbacks
```

jonathanhefner commented on activemodel/test/cases/callbacks_test.rb:
```diff
@@ -55,33 +55,33 @@ def create
     model.create
-    assert_equal model.callbacks, [ :before_create, :before_around_create,
-                                    :create, :after_around_create]
+    assert_equal([ :before_create, :before_around_create,
+                                    :create, :after_around_create], model.callbacks)
```
Indentation is a bit odd here too.  How about:
```suggestion
    assert_equal \
      [:before_create, :before_around_create, :create, :after_around_create],
      model.callbacks
```

jonathanhefner commented on activerecord/test/cases/adapters/postgresql/connection_test.rb:
```diff
@@ -211,8 +211,7 @@ def test_release_non_existent_advisory_lock
         released_non_existent_lock = @connection.release_advisory_lock(fake_lock_id)
-        assert_equal released_non_existent_lock, false,
-          "expected release_advisory_lock to return false when there was no lock to release"
+        assert_equal(false, released_non_existent_lock, "expected release_advisory_lock to return false when there was no lock to release")
```
```suggestion
        assert_equal false, released_non_existent_lock,
          "expected release_advisory_lock to return false when there was no lock to release"
```

jonathanhefner commented on guides/source/command_line.md:
```diff
@@ -106,6 +106,7 @@ default: &default
   adapter: postgresql
   encoding: unicode
 
+
```
:scissors: 

```suggestion
```

jonathanhefner commented on railties/test/application/configuration_test.rb:
```diff
@@ -3807,16 +3807,14 @@ class Post < ActiveRecord::Base
-      assert_equal ActiveStorage.video_preview_arguments,
-        "-vf 'select=eq(n\\,0)+eq(key\\,1)+gt(scene\\,0.015),loop=loop=-1:size=2,trim=start_frame=1'" \
-        " -frames:v 1 -f image2"
+      assert_equal("-vf 'select=eq(n\\,0)+eq(key\\,1)+gt(scene\\,0.015),loop=loop=-1:size=2,trim=start_frame=1'" \
+        " -frames:v 1 -f image2", ActiveStorage.video_preview_arguments)
```
```suggestion
      assert_equal \
        "-vf 'select=eq(n\\,0)+eq(key\\,1)+gt(scene\\,0.015),loop=loop=-1:size=2,trim=start_frame=1' -frames:v 1 -f image2",
        ActiveStorage.video_preview_arguments
```

jonathanhefner commented on actionpack/test/dispatch/mime_type_test.rb:
```diff
@@ -161,8 +161,8 @@ class MimeTypeTest < ActiveSupport::TestCase
   test "type should be equal to symbol" do
-    assert_equal Mime[:html], "application/xhtml+xml"
-    assert_equal Mime[:html], :html
+    assert_equal("application/xhtml+xml", Mime[:html])
+    assert_equal(:html, Mime[:html])
```
There is a [test failure](https://buildkite.com/rails/rails/builds/99592#018a8888-7ec7-4eca-be4d-15c20e7eebc4/1174-1184) due to this change.

This test is actually testing `Mime::Type#==`, so let's invoke that directly:

```suggestion
    assert_operator Mime[:html], :==, "application/xhtml+xml"
    assert_operator Mime[:html], :==, :html
```

shouichi committed:
Enable Minitest/LiteralAsActualArgument

There are assertions that expected/actual arguments are passed in the
reversed order by mistake. Enabling the LiteralAsActualArgument rule
prevents this mistake from happening.

The existing tests were auto-corrected by rubocop with a bit of
indentation adjustment.

Co-authored-by: Jonathan Hefner <jonathan@hefner.pro>

jonathanhefner commented:
Build failures are unrelated.

Thank you, @shouichi! :repeat:
